ftw/testing/freezer.py

```diff
@@ -67,51 +52,12 @@ def now(tz=None):
         expect(time_class()).call(lambda: new_time).count(0, None)
 
         self.mocker.replay()
-        self.enabled = True
-        transaction.get().addBeforeCommitHook(
-            self.transaction_before_commit_hook)
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        datetime_mock_class = datetime
         self.mocker.restore()
         self.mocker.verify()
-
-        # The labix mocker does only replace pointers to the mock-class which
-        # are in dicts.
-        # This does not change the class of instances of our mock-class.
-        # In order to be able to commit while the time is freezed,
-        # we need to replace those classes manually.
-        # Since we cannot patch the class of existing instances,
-        # we need to construct new instances with the real datetime class and
-        # replace all mock instances.
-        for referrer in gc.get_referrers(datetime_mock_class):
-            if not isinstance(referrer, datetime_mock_class):
-                continue
-            if getattr(referrer, '__class__') != datetime_mock_class:
-                continue
-            replacement = datetime(*referrer.timetuple()[:6] + (
-                referrer.microsecond,))
-            global_replace(referrer, replacement)
-
-        self.enabled = False
-
-    def transaction_before_commit_hook(self):
-        """Since a transaction.commit() will serialize the data, it tries
-        to serialize the datetime mock-class when freezing is active.
-        In order to fix pickling problems while committing we disable the
-        freezing temporarily while committing so that mock-based datetime
-        instances are replaced with real ones.
-        """
-        if not self.enabled:
-            return
-
-        self.__exit__(None, None, None)
-        transaction.get().addAfterCommitHook(
-            self.transaction_after_commit_hook)
-
-    def transaction_after_commit_hook(self, status):
-        self.__enter__()
+        curse(datetime, 'now', self._previous_datetime_now)
```

Comment:
Wouldn't [`forbiddenfruit.reverse()`](https://github.com/clarete/forbiddenfruit#reversing-a-curse) be more appropriate here (assuming it works)?